Docs 3228 scale 22.02 update or create certificates articles (#1534)
truenas/documentation: content/SCALE/SCALEUIReference/Credentials/Certificates/CSRScreensSCALE.md

---
title: "Certificate Signing Requests Screens"
description: "This article provides information on SCALE certificates signing request screens and settings."
weight: 30
aliases:
tags:
 - scalecas
 - scalecsrs
 - scalecertificates
---

{{< toc >}}

The **Certificates** screen includes the **Certificate Signing Requests** widget that displays a list of certificate signing requires (CSRs) configured on the system.

Each CSR listed is a link that opens the **Edit CA** screen for the selected CSR.

![CertificateSigningRequestWidgetNoCSR](/images/SCALE/22.02/CertificateSigningRequestWidgetNoCSR.png "Certificate Signing Request Widget No CSRs")

> The <iconify-icon icon="teenyicons:download-solid"></iconify-icon> download icon downloads the CSR to your server.

Reviewer: Download icon not displaying

**Add** opens the **[Add CSR](#add-csr-wizard-screens)** wizard that steps you through setting up a CSR that certifies the ownership of a public key by the named subject of the certificate. 
The **Certificate Signing Requests** section allows users configure the message(s) the system sends to a registration authority of the public key infrastructure to apply for a digital identity certificate. 

## Add CSR Wizard Screens
The **Add CSR** wizard screens step users through configuring a new certificate signing request (CSR) on TrueNAS SCALE. 
The wizard has five different configuration screens, one for each step in the CA configuration process:

1 **[Identifier and Type](#identifier-and-type-options)**

2 **[Certificate Options](#certificate-options)**

3 **[Certificate Subject](#certificate-subject-options)**

4 **[Extra Constraints](#extra-constraints-options)**

5 **[Confirm Options](#confirm-options)**

### Identifier and Type Options
The **Identifier and Type** settings specify the certificate signing request (CSR) name and whether to create a new CSR or import an existing CSR.     
Users can also select a predefined certificate extension from the **Profiles** dropdown list.

> {{< expand "Click Here for More Information" "v" >}}

Reviewer: ```suggestion
{{< expand "Click Here for More Identifier and Type Options Information" "v" >}}
```

| **Profiles** | Displays if **Certificate Signing Request** is set in **Type**. Select a predefined certificate extension from the dropdown list. Choose a profile that best matches your certificate usage scenario. Options are **Openvpn Server Certificate** and **Openvpn Client Certificate**. |
{{< /expand >}}

### Certificate Options
The **Certificate Options** settings specify the type of private key type to use, the number of bits in the key used by the cryptographic algorithm, and the cryptographic algorithm the CSR uses.

There are no **Certificate Options** settings if **Type** on the **Identifier and Type** screen is set to **Import Certificate Signing Request**. 

The **Key Type** selection changes fields displayed. **RSA** is the default setting in **Key Type**.

> {{< expand "Click Here for More Information" "v" >}}

Reviewer: ```suggestion
{{< expand "Click Here for More Certificate Options Information" "v" >}}
```

| **Key Length** | Required. Displays when **RSA** is selected in **Key Type**. Select the number of bits in the key used by the cryptographic algorithm from the dropdown list. Options are **1024**, **2048** or **4096**. For security reasons, a minimum key length of 2048 is recommended. |
| **Digest Algorithm** | Select the cryptographic algorithm to use from the dropdown list. Options are **SHA1**, **SHA224**, **SHA256**, **SHA384** and **SHA512**. Only change the default **SHA256** if the organization requires a different algorithm. |
| **Lifetime** | Enter the number of days for the lifetime of the CA. |
{{< /expand >}}

### Certificate Subject Settings
The **Certificate Subject** settings lets users define the location, name, and email for the organization using the certificate.    
Users can also enter the system [fully-qualified hostname (FQDN)](https://kb.iu.edu/d/aiuv) and any additional domains for multi-domain support.

The **Certificate Subject** settings do not display if **Type** on the **Identifier and Type** screen is set to **Import Certificate Signing Request**. 

> {{< expand "Click Here for More Information" "v" >}}

Reviewer: ```suggestion
{{< expand "Click Here for More Certificate Subject Settings Information" "v" >}}
```

![AddCSRCertificateSubject](/images/SCALE/22.02/AddCSRCertificateSubject.png "Add CSR Certificate Subject CSR Type") 

| Setting | Description |
|---------|-------------|
| **Country** | Required. Select the country of the organization from the dropdown list. |
| **State** | Required. Enter the state or province of the organization. |
| **Locality** | Required. Enter the location of the organization. For example, the city. |
| **Organization** | Required. Enter the name of the company or organization. |
| **Organizational Unit** | Enter the organizational unit of the entity. |
| **Email** | Required. Enter the email address of the person responsible for the CA. |
| **Common Name** | Enter the [fully qualified host name (FQHN)](https://kb.iu.edu/d/aiuv) of the system. This name must be unique within a certificate chain. |

> | **Subject Alternate Names** | Required. Enter additional domains to secure for multi-domain support. Separate each domain by pressing <kbd>Enter</kbd>. For example, if the primary domain is *example.com*, entering *www.example.com** secures both addresses. |

Reviewer: ```suggestion
| **Subject Alternate Names** | Required. Enter additional domains to secure for multi-domain support. Separate each domain by pressing <kbd>Enter</kbd>. For example, if the primary domain is *example.com*, entering *www.example.com* secures both addresses. |
```

{{< /expand >}}

### Extra Constraints Settings
The **Extra Constraints** settings contains certificate extension options:
* **Basic Constraints** that when enabled limits the path length for a certificate chain.
* **Authority Key Identifier** that when enabled provides a means of identifying the public key corresponding to the private key used to sign a certificate.

> * **Key Usage** that when enable defines the purpose of the public key contained in a certificate.
> * **Extended Key Usage** that when enable to further refines key usage extensions.

Reviewer: ```suggestion
* **Key Usage** that when enabled defines the purpose of the public key contained in a certificate.
* **Extended Key Usage** that when enabled further refines key usage extensions.
```

The **Extra Constraints** settings change based on the selection in **Type** on the **Identifier and Type** screen. 

#### Extra Constraints - Certificate Signing Request Type
After selecting **Basic Constraints**, **Authority Key Identifier**, **Extended Key Usage**, or **Key Usage**, each displays more settings that option needs.

> {{< expand "Click Here for More Information" "v" >}}

Reviewer: ```suggestion
{{< expand "Click Here for More Extra Constraints -Certificate Signing Request Type Information" "v" >}}
```

| **Key Usage Config** | Displays after selecting **Extended Key Usage** or **Key Usage**. Select the key usage extension from the dropdown list. Options are **Digital Signature**, **Content Commitment**, **Key Encipherment**, **Data Encipherment**, **Key Agreement**, **Key Cert Sign**, **CRL Sign**, **Encipher Only**, **Decipher Only** or **Critical Extension**. Web certificates typically need at least Digital Signature and possibly Key Encipherment or Key Agreement, while other applications may need other usages. |
{{< /expand >}}

#### Extra Constraints - Import Certificate Signing Request Type
When **Type** on **Identifier and Type** is set to **Import Certificate Signing Request** the **Extra Constraints** screen does not include the options to set extension types.

> {{< expand "Click Here for More Information" "v" >}}

Reviewer: ```suggestion
{{< expand "Click Here for More Extra Constraints -Import Certificate Signing Request Type Information" "v" >}}
```
